### DISCUSSION TITLE
Value objects não deveriam ser imutaveis?

### DISCUSSION THREAD

**Discussion by Gml10** [balta-io/forum#371] [Tira Dúvidas] [answered]
Neste artigo (https://balta.io/blog/ef-core-value-objects) onde mostra o exemplo de um value object "email " há métodos que alteram a propriedade do value object. Mas o conceito de um value object não é que ele é imutavel? Ou seja depois de criado não se deve alterar mais suas propriedades. A abordagem mostrada no artigo esta correta? Não seria melhor implementar essas regras de negocio dentro do proprio entity?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Gml10, sim, a recomendação é que os VOs sejam imutáveis.
Se não necessidade de métodos que façam alteração em seus valores e se o curso de ter um novo objeto representando as mudanças é menor do que alterar, mantemos esta regra sim.

Embora eu nem sempre sigo ela completamente!
